### DISCUSSION TITLE
Finding a specific colour in multiple images and replacing them

### DISCUSSION THREAD

**Discussion by martinum4** [ImageMagick/ImageMagick#4509] [Digital Image Processing]
Hello,

I'd differentiate between a productive and a test setup, they contain the same data and the same general layout.
Since i don't want to check the URLs all the time i'd like to replace the color via script every time i set up the testing system again.
File formats would be .ico, .png and .svg and are logos and icons.

I tired to work with https://legacy.imagemagick.org/discourse-server/viewtopic.php?t=17897 but i'm out of luck.

Can someone help me? Thanks in advance!

**Comment by snibgo** [comment]
Your question is vague.

What gives you problems? I suggest you try replacing a colour in one file. Can you do that? Then put it in a shell loop to do all the files.

What version of IM, on what platform?

**Reply by martinum4** [reply]
Hello, thanks for your reply.
I'm using ImageMagick 6.9.11-60 on Debian 11.
I tried to work my way from before mentioned forum post via `mogrify ./*.png -format png -fill "#333333" -opaque "#C0C0C0" -verbose ./*.png`, However nothing changes and i don't get any error messages, only a list of the images in that directory in the format of 
```
./1.png PNG 48x48 48x48+0+0 8-bit Gray 1092B 0.000u 0:00.000
./1.png PNG 48x48 48x48+0+0 8-bit sRGB 153c 1092B 0.000u 0:00.001
```

**Comment by snibgo** [comment]
You seem to have two files with the same name, which is strange.

Your command syntax is wrong: you give the input names "./*.png" twice. Remove the first one. For mogrify syntax, see https://imagemagick.org/script/mogrify.php

Do any images contain the colour #c0c0c0? If the command has worked and changed images, then no images will contain that colour, so running the command again will not change any pixels.
